src/module/dialogs/combat/CombatDefenseDialog.js

```diff
@@ -225,172 +440,293 @@ export class CombatDefenseDialog extends FormApplication {
+        formula = formula.replace('1d100xa', '0');
+      }
+      if (magicProjection.base >= 200) {
+        // Mastery reduces the fumble range
+        formula = formula.replace('xa', 'xamastery');
+      }
+      const roll = new ABFFoundryRoll(formula, this.attackerActor.system);
+      roll.roll();
 
-          const { spells } = this.defenderActor.system.mystic;
+      if (this.modalData.defender.showRoll) {
+        const flavor = i18n.format('macros.combat.dialog.magicDefense.title', {
+          spell: spell.name,
+          target: this.modalData.attacker.token.name
+        });
 
-          const spell = spells.find(w => w._id === spellUsed);
+        roll.toMessage({
+          speaker: ChatMessage.getSpeaker({ token: this.modalData.defender.token }),
+          flavor
+        });
+      }
 
-          const flavor = i18n.format('macros.combat.dialog.magicDefense.title', {
-            spell: spell.name,
-            target: this.modalData.attacker.token.name
-          });
+      const rolled = roll.total - magicProjection.final - combatModifier;
 
-          roll.toMessage({
-            speaker: ChatMessage.getSpeaker({ token: this.modalData.defender.token }),
-            flavor
-          });
+      this.hooks.onDefense({
+        type: 'mystic',
+        values: {
+          modifier: combatModifier,
+          magicProjection: magicProjection.final,
+          spellGrade,
+          spellUsed,
+          spellName: spell.name,
+          at: at.final,
+          roll: rolled,
+          total: roll.total,
+          spellCasting,
+          supShield,
+          defenderCombatMod
         }
+      });
 
-        const rolled = roll.total - magicProjection - (modifier ?? 0);
-
-        this.hooks.onDefense({
-          type: 'mystic',
-          values: {
-            modifier,
-            magicProjection,
-            spellGrade,
-            spellUsed,
-            at: at.final,
-            roll: rolled,
-            total: roll.total
-          }
-        });
-
-        this.modalData.defenseSent = true;
+      this.modalData.defenseSent = true;
 
-        this.render();
-      }
+      this.render();
     });
 
-    html.find('.send-psychic-defense').click(() => {
-      const { psychicProjection, psychicPotential, powerUsed, modifier } =
-        this.modalData.defender.psychic;
-      const { at } = this.modalData.defender.combat;
+    html.find('.send-psychic-defense').click(async () => {
+      const {
+        psychic: {
+          psychicPotential,
+          powerUsed,
+          modifier,
+          supernaturalShield: { shieldUsed, newShield }
+        },
+        combat: { at },
+        blindnessPen
+      } = this.modalData.defender;
+      const { i18n } = game;
+      const { psychicPowers } = this.defenderActor.system.psychic;
+      const { supernaturalShields } = this.defenderActor.system.combat;
+      let power,
+        fatigue,
+        supShield = { create: false },
```

Comment:
Creo que no se usa esta asignación, porque se sobrescribe cuando redefines más abajo
```suggestion
        supShield,
```

Reply:
Listo